### PR TITLE
Form field to add refill onto Google Calendar 

### PR DESCRIPTION
**Pull request by cartothemax**
This pull request will add a form field for users to switch on weather they want to add medication refills to Google Calendar (for those that used omniauth to join).

Originally users that used omniauth to sign up would automatically have medication refills added their Google Calendar.  This now adds them only if the user has checked the "Add to Google Calendar" checkbox.  

This is my first pull request to the if-me project, so, let me know if I missed anything when doing this work.  Thanks!

### CODE REVIEW

File: app/controllers/medications_controller.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ def create
     respond_to do |format|
       if @medication.save
         # Save refill date to Google calendar
-        if (!current_user.token.blank?)
+        if (!current_user.token.blank?) && params[:add_to_google_cal]
```
Comment (stellacotton) [inline app/controllers/medications_controller.rb]: Could you use the helper `google_oauth2_enabled?` here?

File: app/views/medications/_form.html.erb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <%= form_for(@medication) do |f| %>
 <% if @medication.errors.any? %>
-  <div class="error_explanation">
-    <%= t('.error_explanation') %>
-  </div>
+<div class="error_explanation">
```
Comment (stellacotton) [inline app/views/medications/_form.html.erb]: Why remove the whitespace here? It seems like it's harder to understand that the div is only populated conditionally when it's at the same level as the `if` block.

**Comment by stellacotton** [comment]
@cartothemax thanks for picking this up! Other than a few comments, it looks good so far. Can you add specs to the `adding_medication_spec` or add a controller spec? I know its a bit tricky with the oauth & calendar update stubbing so happy to talk through it or help in any way I can.

File: app/views/medications/_form.html.erb (hunk 2 above)
Comment (cartothemax) [inline app/views/medications/_form.html.erb]: Agreed that does make it more difficult.  I believe my editor did that formatting on it's own.

**Comment by cartothemax** [comment]
@stellacotton thanks for the quick feedback.  I have made the corrections that you pointed out and I will get a spec added.  Do you know of any good examples I could build from what make use of the oauth & calendar update stubbing?

File: app/views/medications/_form.html.erb [hunk 3]
```diff
@@ -93,34 +93,52 @@
...
-       <%= reminder.check_box(:active, { id: "refill_reminder" }, 'true', 'false') %>
-      <div class="label">
-        <%= reminder.label t('.refill_reminder'), for: 'refill_reminder' %>
-        <div class="align_right">
-          <span class="yes_title" title="We'll email you a reminder the week before your medication needs to be refilled"><i class="fa fa-question-circle"></i></span>
-        </div>
+
+    <div class="field">
+      <%= f.fields_for :refill_reminder do | reminder | %>
+      <div class="inline-block">
```
Comment (stellacotton) [inline app/views/medications/_form.html.erb]: I'd indent this as well

File: app/views/medications/_form.html.erb [hunk 4]
```diff
@@ -93,34 +93,52 @@
...
+      <% end %>
+    </div>
 
     <div class="field">
       <%= f.fields_for :take_medication_reminder do | reminder | %>
-        <div class="inline-block">
-          <%= reminder.check_box(:active, { id: 'take_medication_reminder' }, 'true', 'false') %>
-         <div class="label">
-           <%= reminder.label t('.daily_reminder'), for: 'take_medication_reminder' %>
+      <div class="inline-block">
```
Comment (stellacotton) [inline app/views/medications/_form.html.erb]: One more :D

File: app/views/medications/_form.html.erb [hunk 5]
```diff
@@ -93,34 +93,52 @@
...
-         </div>
-         <div class="clear"></div>
         </div>
+        <div class="clear"></div>
+      </div>
       <% end %>
     </div>
+
+    <% if current_user.google_oauth2_enabled? %>
+    <div class="field">
```
Comment (stellacotton) [inline app/views/medications/_form.html.erb]: I think this is the last one

**Comment by stellacotton** [comment]
@cartothemax Hmm, the combo of Google auth + Calendar uploader api calls definitely make it pretty tricky. You could do a feature spec for now that stubs out that the user under test has oauth enabled and stubs out the call to the `CalendarUploader` service (the ifme wrapper around the calendar uploader API)?

I think this is probably a good opportunity to start using VCR to record the flow, but it may be outside the scope of this PR.

**Comment by stellacotton** [comment]
Hey @cartothemax- let me know if you have any other questions about the testing for this PR.

**Comment by cartothemax** [comment]
Hey @stellacotton no, I don't I just need to make the time to get it finished up.  My plan is to finish it this weekend.  Sorry about the delay getting it finished up.

**Comment by stellacotton** [comment]
@cartothemax no worries, give me an @mention if you have any questions or whenever you are ready for a review! :D

**Comment by julianguyen** [comment]
Any updates @cartothemax ? :)

**Comment by julianguyen** [comment]
Let me know when this needs review @cartothemax! ^_^

**Comment by cartothemax** [comment]
@julianguyen I believe I have this in a good spot now although I don't believe that I am correctly stubbing out the Google Calendar call 😞

**Comment by julianguyen** [comment]
@cartothemax What happens when you run the test? Does it stub out the Calendar call at all?

**Comment by julianguyen** [comment]
@cartothemax Hey can you rebase, I'll merge this and continue looking it :)

**Comment by julianguyen** [comment]
Hey @cartothemax, do you mind rebasing this? I can continue to take a look at this once it's merged.

**Comment by cartothemax** [comment]
@julianguyen getting this rebased now.  Sorry for the horrible delay.

**Comment by cartothemax** [comment]
I need to close this pull request and open a newly created pull request off a more up to date version of master.  My rebase is all messed up 😞

**Comment by julianguyen** [comment]
Oh no! Not a problem!

**Pull request closed without merging**